src/rendergraph/common/rendergraph/uniformscache.cpp

```diff
@@ -0,0 +1,27 @@
+#include "rendergraph/uniformscache.h"
+#include "rendergraph/uniformset.h"
+
+using namespace rendergraph;
+
+UniformsCache::UniformsCache(const UniformSet& uniformSet) {
+    int offset = 0;
+    for (const auto& uniform : uniformSet.uniforms()) {
+        const int size = sizeOf(uniform.m_type);
+        m_infos.push_back({uniform.m_type, offset});
```

Comment:
unless you're dealing with return values from `std::make_*` functions, you'll probably want to `emplace_back` instead to avoid the move

Reply:
struct Info has const members, so emplace_back won't work.  I know you will suggest I make them non const and add accessors, but I prefer the simplicity of the struct with const members here.

Reply:
actually thats not what's preventing you from using `emplace_back`. for `emplace_back` you'll need to add a constructor to call with the forwarded arguments. In this case that probably makes sense because otherwise you _will_ get a temporary and a full copy instead of a temporary+move (which you would usually get if you called `push_back(element&&)`. If you add the ctor and use `emplace_back`, you will get the most efficient code (no temporary, move or copy, just constructed the value right where it needs to be in the container).

I'm not saying that we should start optimizing here, just wanted to share my own findings while looking into this.